include/wsl/stubs/rpcndr.h

```diff
@@ -3,10 +3,61 @@
+template<typename T> constexpr const GUID &__wsl_stub_uuidof();
+#define __CRT_UUID_DECL(type, l, w1, w2, b1, b2, b3, b4, b5, b6, b7, b8) \
+    extern "C++"                                                         \
+    {                                                                    \
+        template <>                                                      \
+        struct __wsl_stub_uuidof_s<type>                                 \
+        {                                                                \
+            static constexpr IID __uuid_inst = {                         \
+                l, w1, w2, {b1, b2, b3, b4, b5, b6, b7, b8}};            \
+        };                                                               \
+        template <>                                                      \
+        constexpr const GUID &__wsl_stub_uuidof<type>()                  \
+        {                                                                \
+            return __wsl_stub_uuidof_s<type>::__uuid_inst;               \
+        }                                                                \
+        template <>                                                      \
+        constexpr const GUID &__wsl_stub_uuidof<type *>()                \
+        {                                                                \
+            return __wsl_stub_uuidof_s<type>::__uuid_inst;               \
+        }                                                                \
+    }
+#else
+template<typename T> const GUID &__wsl_stub_uuidof();
+#define __CRT_UUID_DECL(type, l, w1, w2, b1, b2, b3, b4, b5, b6, b7, b8)                  \
+    extern "C++"                                                                          \
+    {                                                                                     \
+        template <>                                                                       \
+        inline const GUID &__wsl_stub_uuidof<type>()                                      \
+        {                                                                                 \
+            static const IID __uuid_inst = {l, w1, w2, {b1, b2, b3, b4, b5, b6, b7, b8}}; \
+            return __uuid_inst;                                                           \
+        }                                                                                 \
+        template <>                                                                       \
+        inline const GUID &__wsl_stub_uuidof<type *>()                                    \
+        {                                                                                 \
+            return __wsl_stub_uuidof<type>();                                             \
+        }                                                                                 \
+    }
+#endif
+#define __uuidof(type) __wsl_stub_uuidof<__typeof(type)>()
```

Comment:
OK, I split first three to merge first.

Reply:
> `__typeof(type)` is a GNU language extension, not core C or C++.

Any better way to do that? in C++14?

Reply:
Oh but thinking about it more, this is in one of the stub headers, i.e. only included when there's no Windows header set... yeah this is actually probably okay since all Linux compilers support these extensions.